apps/accounts/serializers.py

```diff
@@ -20,15 +20,16 @@ class Meta:
             "last_name",
             "password",
         )
-        read_only_fields = ("email", "username")
+
+        read_only_fields = ("email",)
 
 
 class ProfileSerializer(UserDetailsSerializer):
     """
     Serializer to update the user profile.
     """
 
-    affiliation = serializers.CharField(source="profile.affiliation")
+    affiliation = serializers.CharField(source="profile.affiliation", allow_blank=True)
```

Comment:
Why do we allow blank affiliation?

Reply:
This function is called for each instance of dialog we open , so we must allow it or we will have to make a completely separate function for this. in the earlier version we were editing everything in one place so we could make it a must but in this it isn't possible without a view function